src/basicfuns.jl

```diff
@@ -196,16 +201,15 @@ end
 
 Return `log(exp(x) + exp(y))`, avoiding intermediate overflow/undeflow, and handling non-finite values.
 """
-function logaddexp(x::T, y::T) where T<:Real
-    # x or y is  NaN  =>  NaN
-    # x or y is +Inf  => +Inf
-    # x or y is -Inf  => other value
-    isfinite(x) && isfinite(y) || return max(x,y)
-    x > y ? x + log1p(exp(y - x)) : y + log1p(exp(x - y))
-end
-logaddexp(x::Real, y::Real) = logaddexp(promote(x, y)...)
+logaddexp(x::Real, y::Real) = max(x,y) + log1pexp(-ifelse(x == y, zero(x - y), abs(x - y)))
```

Comment:
this is a lot for a one line function. Also would be good to add a comment as for the reason for the `ifelse`.

Reply:
The `ifelse` is for the case `x = y = Inf`, for which we want `x - y` to be 0. I rewrote this and should be clearer now.